internal/logger/component.go

```diff
@@ -199,3 +211,62 @@ func SerializeConnection(conn Connection, extraKeysAndValues ...interface{}) []i
 
 	return keysAndValues
 }
+
+// Server contains data that all server messages MAY contain.
+type Server struct {
+	DriverConnectionID uint64             // Driver's ID for the connection
+	TopologyID         primitive.ObjectID // Driver's unique ID for this topology
+	Message            string             // Message associated with the topology
+	ServerConnectionID *int64             // Server's ID for the connection
+	ServerHost         string             // Hostname or IP address for the server
+	ServerPort         string             // Port for the server
+}
+
+// SerializeServer serializes a Server message into a slice of keys and
+// values that can be passed to a logger.
+func SerializeServer(srv Server, extraKV ...interface{}) []interface{} {
+	// Initialize the boilerplate keys and values.
+	keysAndValues := KeyValues{
+		KeyDriverConnectionID, srv.DriverConnectionID,
+		KeyMessage, srv.Message,
+		KeyServerHost, srv.ServerHost,
+		KeyTopologyID, srv.TopologyID,
+	}
+
+	if connID := srv.ServerConnectionID; connID != nil {
+		keysAndValues.Add(KeyServerConnectionID, connID)
+	}
+
+	port, err := strconv.ParseInt(srv.ServerPort, 0, 32)
```

Comment:
Automatically determining the number base (i.e. base 0) may lead to unexpected results when parsing some port numberes (see examples [here](https://go.dev/play/p/fPJi0XnaMX7)). It's safer to explicitly specify the expected base.
```suggestion
	port, err := strconv.ParseInt(srv.ServerPort, 10, 32)
```

Alternatively, consider using the simpler [strconv.Atoi](https://pkg.go.dev/strconv#Atoi) unless you need to specify the base or bit size.

Reply:
This makes sense, note that the specifications defined the port to be an integer (e.g. [CMAP](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#common-fields)). The two cases you note that would cause unexpected behavior ("0123", "0999") would result in invalid ports regardless of the base